include/up-cpp/transport/builder/UAttributesBuilder.h

```diff
@@ -55,85 +68,168 @@ class UAttributesBuilder {
+         * @return The UAttributes object with the set attributes.
+         */
         uprotocol::v1::UAttributes build() const {
             return attributes_;
         }
 
+        /**
+         * Static factory method that builds a publish message with the given source and priority.
+         * @param source The source URI of the message.
+         * @param priority The priority of the message.
+         * @return A new UAttributesBuilder instance.
+         */
         static UAttributesBuilder publish(const uprotocol::v1::UUri& source, uprotocol::v1::UPriority priority) {
             auto uuid = uprotocol::uuid::Uuidv8Factory::create();
             UAttributesBuilder inst(source, uuid, uprotocol::v1::UMessageType::UMESSAGE_TYPE_PUBLISH, priority);
             return inst;
         }
 
+        /**
+         * Static factory method that builds a notification message with the given source, sink, and priority.
+         * @param source The source URI of the message.
+         * @param sink The sink URI of the message.
+         * @param priority The priority of the message.
+         * @return A new UAttributesBuilder instance.
+         */
         static UAttributesBuilder notification(const uprotocol::v1::UUri& source, const uprotocol::v1::UUri& sink, uprotocol::v1::UPriority priority) {
             auto uuid = uprotocol::uuid::Uuidv8Factory::create();
-            UAttributesBuilder inst(source, uuid, uprotocol::v1::UMessageType::UMESSAGE_TYPE_PUBLISH, priority);
+            UAttributesBuilder inst(source, uuid, uprotocol::v1::UMessageType::UMESSAGE_TYPE_NOTIFICATION, priority);
             inst.setSink(sink);
             return inst;
         }
 
+        /**
+         * Static factory method that builds a request message with the given source, sink, priority, and ttl.
+         * @param source The source URI of the message.
+         * @param sink The sink URI of the message.
+         * @param priority The priority of the message.
+         * @param ttl The time-to-live of the message.
+         * @return A new UAttributesBuilder instance.
+         */
         static UAttributesBuilder request(const uprotocol::v1::UUri& source, const uprotocol::v1::UUri& sink, uprotocol::v1::UPriority priority, int32_t ttl) {
             auto uuid = uprotocol::uuid::Uuidv8Factory::create();
             UAttributesBuilder inst(source, uuid, uprotocol::v1::UMessageType::UMESSAGE_TYPE_REQUEST, priority);
             inst.setSink(sink).setTTL(ttl);
             return inst;
         }
 
+        /**
+         * Static factory method that builds a response message with the given source, sink, priority, and uuid.
+         * @param source The source URI of the message.
+         * @param sink The sink URI of the message.
+         * @param priority The priority of the message.
+         * @param uuid The unique identifier of the message.
+         * @return A new UAttributesBuilder instance.
+         */
```

Comment:
Minor: C++ style blocks (`///`) would be preferred

```suggestion
         /// Static factory method that builds a response message with the given source, sink, priority, and uuid.
         /// @param source The source URI of the message.
         /// @param sink The sink URI of the message.
         /// @param priority The priority of the message.
         /// @param uuid The unique identifier of the message.
         /// @return A new UAttributesBuilder instance.
```

Reply:
done.